File: comic_editor.py

```python
#!/usr/bin/env python3
# comic_editor.py
import argparse
import sys
from PIL import Image, ImageDraw, ImageFont, ImageFilter
from PIL.Image import Resampling

class ComicEditor:
    def __init__(self, input_path, text, x, y, shape='circle', width=200, height=100,
                 bubble_color='#FFFFFF', border_color='#000000', text_color='#000000',
                 font_size=16, output='comic.png'):
        self.input_path = input_path
        self.text = text
        self.x = x
        self.y = y
        self.shape = shape
        self.width = width
        self.height = height
        self.bubble_color = bubble_color
        self.border_color = border_color
        self.text_color = text_color
        self.font_size = font_size
        self.output = output

    def _draw_bubble(self, draw):
        x, y = self.x, self.y
        w, h = self.width, self.height
        if self.shape == 'circle':
            draw.ellipse([x, y, x+w, y+h], fill=self.bubble_color, outline=self.border_color, width=2)
        elif self.shape == 'square':
            draw.rectangle([x, y, x+w, y+h], fill=self.bubble_color, outline=self.border_color, width=2)
        elif self.shape == 'cloud':
            # Простая облачная форма: несколько перекрывающихся кругов
            draw.ellipse([x, y, x+w, y+h//2], fill=self.bubble_color, outline=self.border_color, width=2)
            draw.ellipse([x+w//4, y+h//4, x+w//4+w, y+h//4+h], fill=self.bubble_color, outline=self.border_color, width=2)
            draw.ellipse([x-w//4, y+h//4, x-w//4+w, y+h//4+h], fill=self.bubble_color, outline=self.border_color, width=2)
        return draw

    def run(self):
        try:
            img = Image.open(self.input_path).convert('RGB')
        except Exception as e:
            print(f"Error loading image: {e}", file=sys.stderr)
            sys.exit(1)

        draw = ImageDraw.Draw(img)
        self._draw_bubble(draw)

        try:
            font = ImageFont.truetype("arial.ttf", self.font_size)
        except:
            font = ImageFont.load_default()

        # Центрируем текст в облачке
        bbox = draw.textbbox((0, 0), self.text, font=font)
        text_w = bbox[2] - bbox[0]
        text_h = bbox[3] - bbox[1]
        tx = self.x + (self.width - text_w) // 2
        ty = self.y + (self.height - text_h) // 2
        draw.text((tx, ty), self.text, fill=self.text_color, font=font)

        img.save(self.output)
        print(f"Comic saved to {self.output}")

def main():
    parser = argparse.ArgumentParser(description="Comic Editor (speech bubbles)")
    parser.add_argument("--input", required=True, help="Input image file")
    parser.add_argument("--text", required=True, help="Text in the bubble")
    parser.add_argument("--x", type=int, default=50, help="X coordinate")
    parser.add_argument("--y", type=int, default=50, help="Y coordinate")
    parser.add_argument("--shape", choices=["circle", "square", "cloud"], default="circle")
    parser.add_argument("--width", type=int, default=200)
    parser.add_argument("--height", type=int, default=100)
    parser.add_argument("--bubble-color", default="#FFFFFF")
    parser.add_argument("--border-color", default="#000000")
    parser.add_argument("--text-color", default="#000000")
    parser.add_argument("--font-size", type=int, default=16)
    parser.add_argument("--output", default="comic.png")
    args = parser.parse_args()

    editor = ComicEditor(
        input_path=args.input,
        text=args.text,
        x=args.x,
        y=args.y,
        shape=args.shape,
        width=args.width,
        height=args.height,
        bubble_color=args.bubble_color,
        border_color=args.border_color,
        text_color=args.text_color,
        font_size=args.font_size,
        output=args.output
    )
    editor.run()

if __name__ == "__main__":
    main()

File: comic_editor.go
// comic_editor.go
package main

import (
	"flag"
	"fmt"
	"os"

	"github.com/fogleman/gg"
)

type ComicEditor struct {
	inputPath   string
	text        string
	x, y        int
	shape       string
	width, height int
	bubbleColor string
	borderColor string
	textColor   string
	fontSize    float64
	output      string
}

func NewComicEditor(input, text string, x, y int, shape string, w, h int,
	bubble, border, textColor string, fontSize float64, output string) *ComicEditor {
	return &ComicEditor{
		inputPath:   input,
		text:        text,
		x:           x,
		y:           y,
		shape:       shape,
		width:       w,
		height:      h,
		bubbleColor: bubble,
		borderColor: border,
		textColor:   textColor,
		fontSize:    fontSize,
		output:      output,
	}
}

func (e *ComicEditor) Run() error {
	// Load image
	dc, err := gg.LoadImage(e.inputPath)
	if err != nil {
		return err
	}
	ctx := gg.NewContextForImage(dc)

	// Draw bubble
	x, y := float64(e.x), float64(e.y)
	w, h := float64(e.width), float64(e.height)
	ctx.SetHexColor(e.bubbleColor)
	ctx.SetHexColor(e.borderColor)

	switch e.shape {
	case "circle":
		ctx.DrawEllipse(x+w/2, y+h/2, w/2, h/2)
		ctx.FillPreserve()
		ctx.SetLineWidth(2)
		ctx.Stroke()
	case "square":
		ctx.DrawRectangle(x, y, w, h)
		ctx.FillPreserve()
		ctx.SetLineWidth(2)
		ctx.Stroke()
	case "cloud":
		ctx.DrawEllipse(x+w/3, y+h/3, w/3, h/3)
		ctx.FillPreserve()
		ctx.SetLineWidth(2)
		ctx.Stroke()
		ctx.DrawEllipse(x+w*2/3, y+h/3, w/3, h/3)
		ctx.FillPreserve()
		ctx.SetLineWidth(2)
		ctx.Stroke()
		ctx.DrawEllipse(x+w/2, y+h*2/3, w/3, h/3)
		ctx.FillPreserve()
		ctx.SetLineWidth(2)
		ctx.Stroke()
	}

	// Draw text
	ctx.SetHexColor(e.textColor)
	if err := ctx.LoadFontFace("arial.ttf", e.fontSize); err != nil {
		ctx.LoadFontFace("/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf", e.fontSize)
	}
	ctx.DrawStringAnchored(e.text, x+w/2, y+h/2, 0.5, 0.5)

	// Save
	return ctx.SavePNG(e.output)
}

func main() {
	var (
		input       string
		text        string
		x, y        int
		shape       string
		width, height int
		bubbleColor string
		borderColor string
		textColor   string
		fontSize    float64
		output      string
	)
	flag.StringVar(&input, "input", "", "Input image file")
	flag.StringVar(&text, "text", "", "Text in the bubble")
	flag.IntVar(&x, "x", 50, "X coordinate")
	flag.IntVar(&y, "y", 50, "Y coordinate")
	flag.StringVar(&shape, "shape", "circle", "circle, square, cloud")
	flag.IntVar(&width, "width", 200, "Bubble width")
	flag.IntVar(&height, "height", 100, "Bubble height")
	flag.StringVar(&bubbleColor, "bubble-color", "#FFFFFF", "Bubble color")
	flag.StringVar(&borderColor, "border-color", "#000000", "Border color")
	flag.StringVar(&textColor, "text-color", "#000000", "Text color")
	flag.Float64Var(&fontSize, "font-size", 16, "Font size")
	flag.StringVar(&output, "output", "comic.png", "Output file")
	flag.Parse()

	if input == "" || text == "" {
		fmt.Println("Error: --input and --text are required")
		os.Exit(1)
	}

	editor := NewComicEditor(input, text, x, y, shape, width, height,
		bubbleColor, borderColor, textColor, fontSize, output)
	if err := editor.Run(); err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Comic saved to %s\n", output)
}
